console/src/components/ProfileDrawer/ProfileDrawer.js

```diff
@@ -0,0 +1,191 @@
+import Skinnable from '@enact/agate/Skinnable';
+import Drawer from '@enact/agate/Drawer';
+import {ToggleButtonBase} from '@enact/agate/ToggleButton';
+import {Cell, Column} from '@enact/ui/Layout';
+import {adaptEvent, forward, handle} from '@enact/core/handle';
+import LabeledIconButton from '@enact/agate/LabeledIconButton';
+import {TabGroupBase} from '@enact/agate/Panels/TabbedPanels';
+import kind from '@enact/core/kind';
+import Slottable from '@enact/ui/Slottable';
+import PropTypes from 'prop-types';
+import React from 'react';
+import compose from 'ramda/src/compose';
+
+import UserAvatar from '../UserAvatar';
+import AppContextConnect from '../../App/AppContextConnect';
+import UserSelectionPopup from '../UserSelectionPopup';
+
+import componentCss from './ProfileDrawer.less';
+
+const SlottedTabGroup = Skinnable(Slottable({slots: ['afterTabs', 'beforeTabs']}, TabGroupBase));
+
+const panelIndexes = [
+	'home',
+	'settings',
+	'settings/theme'
+];
+
+const ProfileDrawerBase = kind({
+	name: 'ProfileDrawer',
+
+	propTypes: {
+		getPanelIndexOf: PropTypes.func.isRequired,
```

Comment:
Passing this down seems to invert the responsibilities between ProfileDrawer and its consumer since ProfileDrawer need not know how the selected value maps to panel indices.

Reply:
This awkward implementation is to avoid a circular dependency.

ProfileDrawer needs to know which panel index it's on, to highlight the correct tab when it's on that panel. The collections of tabs (in ProfileDrawer and panels (from the main app) differ, and it needs a way to look up those indexes and essentially map those to the current indexes. I'm open to suggestions for improvement that retain all of the functionality (UX and DX) of the current implementation.

Reply:
Not important enough to fret over, imo.